presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -332,7 +345,51 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
             case GREATER_THAN_OR_EQUAL:
                 return planQuantifiedOrderable(subPlan, quantifiedComparison, correlationAllowed);
         }
-        throw new SemanticException(NOT_SUPPORTED, quantifiedComparison, "Quantified comparison with the given quantifier and comparison type is not supported");
+        // all cases are checked, so this exception should never be thrown
+        throw new IllegalStateException("Unexpected combination of quantifier and comparison type: " +
+                quantifiedComparison.getQuantifier() + ", " + quantifiedComparison.getComparisonType());
+    }
+
+    private PlanBuilder planQuantifiedEqualsAll(PlanBuilder subPlan, QuantifiedComparisonExpression quantifiedComparison, boolean correlationAllowed)
+    {
+        checkArgument(quantifiedComparison.getComparisonType() == EQUAL && quantifiedComparison.getQuantifier() == Quantifier.ALL);
+        // A = ALL B <=> min B = max B && A = min B
+        RelationPlan subqueryRelationPlan = createRelationPlan(quantifiedComparison.getSubquery());
+        PlanNode subqueryPlan = subqueryRelationPlan.getRoot();
+        List<Expression> outputColumnReference = ImmutableList.of(getOnlyElement(subqueryRelationPlan.getOutputSymbols()).toSymbolReference());
+        Type outputColumnType = getOnlyElement(subqueryRelationPlan.getDescriptor().getAllFields()).getType();
+        if (!outputColumnType.isOrderable()) {
+            throwNotSupportedException(quantifiedComparison, "Quantified comparison with given parameters on unorderable type");
```

Comment:
Rephrased the message to include quantified comparison and the type.
I was thinking about using 'count distinct' to implement it for unorderable types, but it would be less efficient than min/max. So, I decided to deal only with orderable types in this patch and will create a follow up task to support =ALL/<>ANY with unorderable types.
What do you think?